community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/util/PathInterestFactory.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphalgo.impl.util;
+
+import java.util.Comparator;
+
+import org.neo4j.function.BiFunction;
+import org.neo4j.kernel.impl.util.Precision;
+
+/**
+ * @author Anton Persson
+ */
+public class PathInterestFactory
+{
+    public static <P extends Comparable<P>> PathInterest<P> singlePathInterest()
```

Comment:
It seems unnecessary to instantiate these PathInterest objects all the time. There should be anonymous generic-neutral private final instances inside this factory and just merely handed out via these generic static accessor methods.